app.py

```diff
@@ -151,19 +172,20 @@ def incoming_event():
                 db.session.add(sender)
                 db.session.commit()
                 sender = User.query.filter_by(slack_id=sender_slack_id).first()
-
+            
             receiver_slack_id = event.get('item_user')
-            receiver = User.query.filter_by(slack_id=receiver_slack_id).first()
-            if receiver is None:
-                user_data = get_user_by_slack_id(receiver_slack_id)
-                receiver = User(
-                    slack_id = receiver_slack_id,
-                    display_name = user_data.get('display_name'),
-                    team_id = team_id
-                )
-                db.session.add(receiver)
-                db.session.commit()
+            if receiver_slack_id:
                 receiver = User.query.filter_by(slack_id=receiver_slack_id).first()
+                if receiver is None:
+                     user_data = get_user_by_slack_id(receiver_slack_id)
+                     receiver = User(
+                         slack_id = receiver_slack_id,
+                         display_name = user_data.get('display_name'),
+                         team_id = team_id
+                     )
+                     db.session.add(receiver)
+                     db.session.commit()
+                     receiver = User.query.filter_by(slack_id=receiver_slack_id).first()
```

Comment:
instead of querying the db again, i think the receiver we already have will be updated once the db is committed, so we can probably remove this line.